src/wp-includes/class-wp-theme.php

```diff
@@ -122,6 +122,14 @@ final class WP_Theme implements ArrayAccess {
 	 */
 	private $block_theme;
 
+	/**
+	 * Theme has block templates.
+	 *
+	 * @since 6.3.0
+	 * @var bool
+	 */
+	private $has_block_templates;
```

Comment:
See my comment above, I think it would be more efficient to cache the actual template folders array instead of whether the theme uses the _old_ folder names.

The variable name and documentation here is also a bit confusing, since it's not about whether the theme has block templates, but rather whether the theme uses the old folder naming convention for block templates.